### ISSUE TITLE
[Bug] Some passwords are not auto-filling

### ISSUE THREAD

**Issue by nine76evil** [mozilla-mobile/fenix#12498]
Save username/passwords and sync between devices

Firefox auto-fills username/passwords into appropriate fields

Firefox auto-fills on desktop but not Android even though it shows in settings > Logins and passwords

Mozilla Firefox Beta 79.0.0-beta.5
Add-ons: Dark Reader, HTTPS Everywhere, uBlock Origin 
Note: Issue persist even with add-ons disabled

Google Pixel 3a Android 10

**Comment by sv-ohorvath** [comment]
I can confirm this: if you sync the passwords they are not auto-filled. If you save a new one (but just one) from the mobile device, you will get that login auto-filled.

**Comment by ekager** [comment]
@sv-ohorvath I can also repro this on certain sites, but for example my synced logins from Github and 15Five do autofill correctly. Are you also noticing that it is site specific?

**Comment by sv-ohorvath** [comment]
@ekager Yes, it seems to be site-specific. In my case, twitter & facebook don't auto-fill the synced passwords, but github and wiki.mozilla.com do.

**Comment by ekager** [comment]
Nice find! I'll have to discuss this with other teams to see what we should be doing here 🤔

**Comment by ekager** [comment]
Okay chatted with the toolkit team and this is currently expected (albeit) frustrating behavior. We don’t want to “autofill” (without interaction) different subdomains for privacy concerns (ie filling your facebook credentials into evil.facebook.com would be bad). On desktop we follow the same behavior and offer a selection dropdown to select logins from likely matching sites. Hopefully once #5551 lands, we can follow the desktop behavior here.

**Comment by sv-ohorvath** [comment]
Since the reporter hasn't responded if he's seeing a different issue, I think we can close this as expected. @nine76evil If you have different issues then what has been discussed please reopen.

**Comment by nine76evil** [comment]
My logins/passwords are still not syncing for certain sites. One example being https://real-debrid.com/ if i open Firefox on my laptop (Linux) it auto-fills and logins me in. However on the mobile browser it doesn't hands and username or password saved.

**Comment by sv-ohorvath** [comment]
Ok, I can also reproduce on that website and it doesn't have a mobile subdomain. cc: @ekager

**Comment by nine76evil** [comment]
I'm also curious why there is no option to copy link location i.e. copying a magnet link by long pressing. This features exist in Android chrome which i really don't want to use.

**Comment by petrst** [comment]
> @sv-ohorvath I can also repro this on certain sites, but for example my synced logins from Github and 15Five do autofill correctly. Are you also noticing that it is site specific?

Github doesn't autofill in my case.

**Comment by LinAGKar** [comment]
Something I noticed is that the autofill seems to break if you have multiple passwords saved for the same site.

**Comment by ekager** [comment]
@LinAGKar that is fixed in Beta/Nightly and will eventually make its way to Release. We not have a way to select from multiple logins.

**Comment by ekager** [comment]
@petrst can't repro with GitHub on Nightly.
These are the STR:

1. Have GitHub saved with logins on desktop
2. Log into FxA and sync logins on Firefox (go into Saved Logins and verify logins have synced)
3. Go to github.com/login 

My login autofilled. Are you still reproducing this? What URL do you have saved for github on desktop?
<img width="268" alt="Screen Shot 2020-09-14 at 5 33 20 PM" src="https://user-images.githubusercontent.com/25442310/93151375-9d2efa00-f6b0-11ea-9115-19eac5880c24.png">

**Comment by ekager** [comment]
> I'm also curious why there is no option to copy link location i.e. copying a magnet link by long pressing. This features exist in Android chrome which i really don't want to use.

@nine76evil please open a new issue for this feature request

**Comment by ekager** [comment]
With a fake saved login for real-debrid.com I am getting the select login prompt which is the same behavior I get on desktop.

**Comment by ekager** [comment]
Okay I think I've investigated all the specific cases here and they are not reproducible for me on Nightly. I'm going to close this ticket because it's pretty confusing. If any of you can still reproduce on a specific site in Nightly or are seeing other weird login behavior, please open a new issue and tag me. Thanks!

**Comment by nine76evil** [comment]
This issue is still existing for me on nightly. Real-debris.com is one example.
